ios/Classes/BetterPlayer.m

```diff
@@ -654,123 +618,54 @@ - (void)setRestoreUserInterfaceForPIPStopCompletionHandler:(BOOL)restore
-                [self setupPipController];
-            }
-        });
-    } else {
-        if (@available(iOS 14.2, *) && _pipController) {
-            _pipController.canStartPictureInPictureAutomaticallyFromInline = false;
+        if (!_pipController) {
+            [self setupPipController];
         }
-    }
-}
 
-- (void)updatePipController {
-    if (@available(iOS 9.0, *)) {
-        [[AVAudioSession sharedInstance] setActive: YES error: nil];
-        [[UIApplication sharedApplication] beginReceivingRemoteControlEvents];
-        AVPlayerLayer *playerLayer = self._betterPlayerView.playerLayer;
-        if (_pipController && playerLayer && [AVPictureInPictureController isPictureInPictureSupported]) {
-            if (_pipController.contentSource.playerLayer != playerLayer) {
-                _pipController.contentSource = [[AVPictureInPictureControllerContentSource alloc] initWithPlayerLayer:playerLayer];
-            } else {
-                // incase other playerLayers are disposed, use the original one
-                if (_originPlayerView.playerLayer != nil) {
-                    _pipController.contentSource = [[AVPictureInPictureControllerContentSource alloc] initWithPlayerLayer:_originPlayerView.playerLayer];
-                }
-            }
-            if (@available(iOS 14.2, *)) {
-                _pipController.canStartPictureInPictureAutomaticallyFromInline = self._willStartPictureInPicture;
-            }
-            _pipController.delegate = self;
+        if (_pipController && !_pipController.canStartPictureInPictureAutomaticallyFromInline) {
+            _pipController.canStartPictureInPictureAutomaticallyFromInline = YES;
         }
-    } else {
-        // Fallback on earlier versions
+    } else if (_pipController && _pipController.canStartPictureInPictureAutomaticallyFromInline) {
+        _pipController.canStartPictureInPictureAutomaticallyFromInline = NO;
     }
 }
 
-- (void)disablePictureInPicture
-{
-    if (__playerLayer){
-        [self._playerLayer removeFromSuperlayer];
-        self._playerLayer = nil;
-        if (_pipController) {
-            _pipController = nil;
-            [self willStartPictureInPicture: true];
-        }
-        if (_eventSink != nil) {
-            _eventSink(@{@"event" : @"pipStop"});
-        }
+- (void)gotoBackgroundWithPIP:(CGRect)frame {
+    [self setBeforePipSourceRectHint:frame];
+
+    if (_pipController && _pipController.canStartPictureInPictureAutomaticallyFromInline) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [[UIApplication sharedApplication] performSelector:@selector(suspend)];
+        });
     }
 }
-#endif
 
-#if TARGET_OS_IOS
 - (void)pictureInPictureControllerDidStopPictureInPicture:(AVPictureInPictureController *)pictureInPictureController  API_AVAILABLE(ios(9.0)){
-    [self disablePictureInPicture];
     if (_eventSink != nil && !isRestorePip) {
         _eventSink(@{@"event" : @"closePip"});
+        [self pause];
```

Comment:
missing send pause event can cause player's control not synced UI (play/pause button)

Reply:
bằng cách thần kỳ nào đó thì em test thì thấy nó vẫn bắn được event pause lên anh ạ :))

Reply:
See the button is still || when press X button from PIP UI

Reply:
em test nó paused và update button thành ▶ rồi anh ạ, nên không cần bắn event lên nữa, để tránh duplicate